src/v0/destinations/braze/transform.js

```diff
@@ -178,6 +181,127 @@ function getUserAttributesObject(message, mappingJson, destination) {
+    const aliasesToIdentify = identifyCallsChunk.flatMap(
+      (identifyCall) => identifyCall.identifyPayload.aliases_to_identify,
+    );
+
+    let brazeIdentifyResp;
+    try {
+      const { processedResponse } = await handleHttpRequest(
+        'post',
+        identifyEndpoint,
+        { aliases_to_identify: aliasesToIdentify, merge_behavior: 'merge' },
+        {
+          headers: {
+            // eslint-disable-next-line sonarjs/no-duplicate-string
+            'Content-Type': 'application/json',
+            Accept: 'application/json',
+            Authorization: `Bearer ${destination.Config.restApiKey}`,
+          },
+        },
+        {
+          destType: 'braze',
+          feature: 'transformation',
+          requestMethod: 'POST',
+          module: 'router',
+          endpointPath: '/users/identify',
+        },
+      );
+      brazeIdentifyResp = processedResponse;
+    } catch (networkError) {
+      logger.error(
+        'Braze identity resolution request failed with possible network error',
+        networkError,
+      );
+      stats.increment('braze_batched_identify_func_calls_count', {
+        destination_id: destinationId,
+        status: '5xx',
+        error: 'request_error',
+      });
+      return new NetworkError(
+        `Braze identify failed - ${networkError.message || 'Network request failed'}`,
+        500,
+        {
+          [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(500),
+        },
+      );
+    }
+
+    if (!isHttpStatusSuccess(brazeIdentifyResp.status)) {
+      logger.error('Braze identity resolution failed', JSON.stringify(brazeIdentifyResp.response));
+      collectStatsForAliasFailure(brazeIdentifyResp.response, destination.ID);
+      stats.increment('braze_batched_identify_func_calls_count', {
+        destination_id: destinationId,
+        status: brazeIdentifyResp.status,
+        error: 'non_2xx_status',
+      });
+      return new NetworkError(
+        `Braze identify failed - ${JSON.stringify(brazeIdentifyResp.response)}`,
+        brazeIdentifyResp.status,
+        {
+          [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(brazeIdentifyResp.status),
+        },
+        brazeIdentifyResp.response,
+      );
+    }
+
+    if (brazeIdentifyResp.response?.errors?.length > 0) {
+      // We want this scenario to throw bugsnag error so that we can fix the issue
+      logger.error(
+        'Unhandled Identify Resolution Error',
+        JSON.stringify(brazeIdentifyResp.response),
+      );
+      stats.increment('braze_batched_identify_func_calls_count', {
+        destination_id: destinationId,
+        status: brazeIdentifyResp.status,
+        error: 'unhandled_error',
+      });
+      throw new Error(
+        `[Unhandled Identify Resolution Error] ${brazeIdentifyResp.response?.errors[0]?.type}`,
+      );
+    }
+
+    return null; // Success
+  });
+  return Promise.all(allRequests);
```

Comment:
Before returning verify no issues with responses and then return null right now the code is bit difficult to understand. If needed we can break this function to sub functions

Reply:
better we do this controlled manner to avoid getting throttled, Promise.all(allRequests) doens't provide any control

Reply:
ok